### PR TITLE
Renew auth token on 401

### PR DESCRIPTION
**Pull request by mustafaturan**
### Description

Renew auth token on 401

**When the response is 401:**
- Auto-renew auth token and update request header
- Retry the requests once more

**Additional changes:**
- Create retriable_auth middleware plug/unplug easily
- Move Faraday middlewares into its own folder
- Register middlewares

### Tasks
- [x] Renew auth token on 401

### CODE REVIEW

File: lib/iron_bank/client.rb [hunk 1]
```diff
@@ -44,8 +44,10 @@ def connection
       @connection ||= Faraday.new(faraday_config) do |conn|
         conn.use      :ddtrace, open_tracing_options if open_tracing_enabled?
         conn.use      instrumenter, instrumenter_options if instrumenter
-        conn.use      IronBank::Response::RaiseError
+        conn.use      :raise_error
```
Comment (mickaelpham) [inline lib/iron_bank/client.rb]: This still loads `IronBank::Response::RaiseError`?

File: lib/iron_bank/client.rb [hunk 2]
```diff
@@ -44,8 +44,10 @@ def connection
...
         conn.request  :json
+        conn.request  :retry, max: 2, retry_statuses: [401]
```
Comment (mickaelpham) [inline lib/iron_bank/client.rb]: make the `max:` an optional configuration?

File: lib/iron_bank/faraday_middleware.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# IronBank main module
+module IronBank
+  # IronBank Faraday middleware module
+  module FaradayMiddleware
+    autoload :RaiseError,    'faraday_middleware/raise_error'
```
Comment (mickaelpham) [inline lib/iron_bank/faraday_middleware.rb]: not a fan of `autoload`, let's remove it and instead add it to the list of `require` in `iron_bank.rb`

cf. https://bugs.ruby-lang.org/issues/5653

> "I strongly discourage the use of autoload in any standard libraries" (Re: autoload will be dead)
(Matz)

Comment (mustafaturan) [reply lib/iron_bank/faraday_middleware.rb]: My bad, I forgot to remove after adding the `require` inside the main module

**Pull request merged**